var/spack/repos/builtin/packages/nest-simulator/package.py

```diff
@@ -0,0 +1,138 @@
+    version('2.12.0', sha256='bac578f38bb0621618ee9d5f2f1febfee60cddc000ff32e51a5f5470bb3df40d')
+    version('2.10.0', sha256='2b6fc562cd6362e812d94bb742562a5a685fb1c7e08403765dbe123d59b0996c')
+    version('2.8.0',  sha256='d47325b27a5599b6ea58a3c4ef06656e7c5a4941c4e94dec6a5c2fa956209915')
+    version('2.6.0',  sha256='5fe4924bc57d0c7dd820aa371de935eedf7e813832c0eee2c976b33c9a8db4cf')
+    version('2.4.2',  sha256='8f86e58c1a12b733ffabd8b0400326e5a3494a458149ea8ebe9f19674d05b91b')
+
+    variant('python', default=True,
+            description='Build the PyNest interface')
+    variant('mpi', default=True,
+            description='Build with MPI bindings')
+    variant('optimize', default=True,
+            description='Build with MPI bindings')
+    variant('modules', default=False,
+            description='Enables external module support')
+    # TODO add variants for neurosim and music when these are in spack
+
+    depends_on('python',            when='+python')
+    depends_on('py-numpy',          when='+python')
+    depends_on('py-cython@0.19.2:', when='+python')
+    depends_on('py-nose',           when='+python')
+    depends_on('py-setuptools',     when='+python')
+
+    depends_on('mpi', when='+mpi')
+
+    depends_on('doxygen')
+
+    depends_on('gsl')
+    depends_on('readline')
+    depends_on('libtool')
+    depends_on('pkg-config')
+
+    extends('python', when='+python')
+
+    # Before 2.12.0 it was an autotools package
+    @when('@:2.10.99')
+    def cmake(self, spec, prefix):
+        pass
+
+    @when('@:2.10.99')
+    def build(self, spec, prefix):
+        pass
+
+    @when('@:2.10.99')
+    def install(self, spec, prefix):
+        configure_args = ["CXXFLAGS=-std=c++03",
+                          "--prefix=" + prefix,
+                          "--with-openmp"]
+        if '+python' in spec:
+            configure_args.append("--with-python")
+        else:
+            configure_args.append("--without-python")
+        if '+mpi' in spec:
+            configure_args.append("--with-mpi")
+        else:
+            configure_args.append("--without-mpi")
+        if '+optimize' in spec:
+            configure_args.append("--with-optimize")
+        else:
+            configure_args.append("--without-optimize")
+
+        configure(*configure_args)
+
+        make()
+        make("install")
+
+    def cmake_args(self):
+        args = []
+        if '+mpi' in self.spec:
+            args.append('-Dwith-mpi=ON')
+        else:
+            args.append('-Dwith-mpi=OFF')
+        if '+python':
+            version = self.spec['python'].version[0]
+            args.append('-Dwith-python={0}'.format(version))
+        else:
+            args.append('-Dwith-python=OFF')
+        if '+optimize':
+            args.append('-Dwith-optimize=ON')
+        else:
+            args.append('-Dwith-optimize=OFF')
+
+        return args
+
+    @when('+modules')
+    @when('@:2.14.0')
```

Comment:
Does this actually work? I think you may want to combine both of these into a single `@when`. Otherwise the latter may override the former.

Reply:
I thought it did, but on one line it's more readable